File: Erlang/0007.erl
-export([reverse/1]).

-spec reverse(X :: integer()) -> integer().
reverse(X) ->
    % Define the 32-bit signed integer range
    MinInt = -2_147_483_648,
    MaxInt = 2_147_483_647,

    % Check if the number is negative
    Negative = X < 0,

    % Work with the absolute value of X
    XAbs = abs(X),

    % Convert to string, reverse, and convert back to integer
    ReversedStr = lists:reverse(integer_to_list(XAbs)),
    ReversedInt = list_to_integer(ReversedStr),

    % Apply the original sign to the reversed integer
    Result = if
        Negative -> -ReversedInt;
        true -> ReversedInt
    end,

    % Check for overflow and return the result
    case Result of
        Result when Result < MinInt; Result > MaxInt -> 0;
        _ -> Result
    end.

File: Elixir/0007.exs
defmodule Solution do
  @spec reverse(x :: integer) :: integer
  def reverse(x) do
    # Define the 32-bit signed integer range
    min_int = -2_147_483_648
    max_int = 2_147_483_647

    # Check if the input is negative
    negative = x < 0

    # Work with the absolute value of x
    x_abs = abs(x)

    # Reverse the digits of the absolute value
    reversed_str = Integer.to_string(x_abs)
    |> String.reverse()
    |> String.to_integer()

    # Apply the original sign to the reversed integer
    reversed_int = if negative, do: -reversed_str, else: reversed_str

    # Check for overflow and return the result
    if reversed_int < min_int or reversed_int > max_int do
      0
    else
      reversed_int
    end
  end
end
